src/pages/guide/integration/data-fixtures-guide.md

```diff
@@ -0,0 +1,673 @@
+    ),
+    DataFixture(
+        BundleProductFixture::class,
+        [
+            'sku' => 'bundle-product',
+            'price' => 50,
+            '_options' => ['$option$']
+        ]
+    )
+]
+public function testBundleProduct(): void
+{
+    // Test bundle product logic
+}
+```
+
+## Fixture scope: Class vs Method
+
+Fixtures can be applied at the test class level or method level:
+
+### Class-level fixtures
+
+```php
+#[
+    DataFixture(ProductFixture::class, count: 10)
+]
+class ProductCollectionTest extends TestCase
+{
+    // All tests in this class will have 10 products available
+    
+    public function testCount(): void { }
+    
+    public function testFilter(): void { }
+}
+```
+
+### Method-level fixtures
+
+```php
+class ProductTest extends TestCase
+{
+    #[DataFixture(ProductFixture::class, ['price' => 10])]
+    public function testCheapProduct(): void { }
+    
+    #[DataFixture(ProductFixture::class, ['price' => 1000])]
+    public function testExpensiveProduct(): void { }
+}
+```
+
+<InlineAlert variant="warning" slots="text" />
+
+When method-level fixtures are present, class-level fixtures are **ignored** for that method.
+
+## Best practices
+
+### 1. Do not create legacy fixtures
+
+<InlineAlert variant="warning" slots="text" />
+
+**New legacy fixtures cannot be created.** Do not create new tests using legacy fixtures (e.g., `Magento/Catalog/_files/product_simple.php`). Always use parametrized data fixtures instead. Existing legacy fixtures should be migrated.
+
+**Bad - Cannot create new:**
+```php
+/**
+ * @magentoDataFixture Magento/Catalog/_files/product_simple.php
+ */
+```
+
+**Good - Use parametrized data fixtures:**
+
+With PHP Attributes:
+```php
+#[DataFixture(ProductFixture::class, ['sku' => 'test-product'])]
+```
+
+Or with DocBlock annotations:
+```php
+/**
+ * @magentoDataFixture \Magento\Catalog\Test\Fixture\Product with:{"sku": "test-product"} as:product
+ */
+```
+
+### 2. Prefer PHP Attributes for new tests
+
+PHP Attributes provide better type safety, IDE support, and are easier to read than the DocBlock annotation `with:` and `as:` syntax.
+
+### 3. Keep fixtures focused
+
+Each fixture should do one thing. Use multiple fixtures and compose them rather than creating complex all-in-one fixtures.
+
+### 4. Use fixture aliases
+
+Always use the `as` parameter for fixtures you'll reference later:
+
+```php
+#[
+    DataFixture(CategoryFixture::class, as: 'category'),  // Good
+    DataFixture(ProductFixture::class, ['category_ids' => ['$category.id$']])
+]
+```
+
+### 5. Leverage dynamic defaults
+
+Most modern fixtures use `%uniqid%` to auto-generate unique values. You don't always need to specify every field:
+
+```php
+#[
+    DataFixture(ProductFixture::class),  // Creates product with unique SKU
+    DataFixture(ProductFixture::class)   // Creates another with different unique SKU
+]
+```
+
+### 6. Use appropriate isolation
+
+Combine fixtures with appropriate isolation attributes:
```

Comment:
DB is isolated by default. AppIsolation should not be recommended without asterisk. It's is not recommended to use it unless necessary. it has performance implications.